File: Activities/dp/fib.r
fib <- function(n) {
    r <- rep(0,  n)

    r[1] <- 0
    r[2] <- 1

    for( i in 3:n ) {
        r[i] <- r[i - 1] + r[i - 2]
    }
    return( r[n] )
}

File: Activities/dp/fib.py
def fib(n: int) -> int:
    r = [0] * (n + 1)

    r[0] = 0
    r[1] = 0
    r[2] = 1

    for i in range(3, n + 1):
        r[i] = r[i - 1] + r[i - 2]

    return r[n]
